doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,498 @@
+
+You can use :ref:`west flash <west-flashing>` to flash applications with
+sysbuild.
+
+When invoking ``west flash`` on a build consisting of multiple images, then each
+image is flashed in sequence. Extra arguments such as ``--runner jlink`` are
+passed to each invocation.
+
+For more details, see :ref:`west-multi-domain-flashing`.
+
+Sysbuild debugging: ``west debug``
+**********************************
+
+You can debug the main application when using sysbuild with ``west debug`` in
+the same way you'd use it without sysbuild. Just follow the existing :ref:`west
+debug <west-debugging>` guide to debug the main sample.
+
+To debug a different domain (Zephyr application), such as ``mcuboot``, use
+the ``--domain`` argument, like this::
+
+  west debug --domain mcuboot
+
+For more details, see :ref:`west-multi-domain-debugging`.
+
+Building a sample with MCUboot
+******************************
+
+Just as Zephyr itself is compatible with MCUboot, sysbuild supports MCUboot
+natively.
+
+To build a sample, such as ``hello_world``, with MCUboot, it is enough to just
+enable MCUboot and build and flash the sample, like this:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      .. zephyr-app-commands::
+         :tool: west
+         :app: samples/hello_world
+         :board: reel_board
+         :goals: build
+         :west-args: --sysbuild
+         :gen-args: -DSB_CONFIG_BOOTLOADER_MCUBOOT=y
+         :compact:
+
+   .. group-tab:: ``cmake``
+
+      .. zephyr-app-commands::
+         :tool: cmake
+         :app: share/sysbuild
+         :board: reel_board
+         :goals: build
+         :gen-args: -DAPP_DIR=samples/hello_world -DSB_CONFIG_BOOTLOADER_MCUBOOT=y
+         :compact:
+
+This builds ``hello_world`` and ``mcuboot`` for the ``reel_board``, and then
+flashes both the ``mcuboot`` and ``hello_world`` application images to the
+board.
+
+More detailed information regarding the use of MCUboot with Zephyr can be found
+in the `MCUboot with Zephyr`_ documentation page on the MCUboot website.
+
+.. note::
+
+   MCUBoot default configuration will perform a full chip erase when flashed.
+   This can be controlled through the MCUBoot Kconfig option
+   ``CONFIG_ZEPHYR_TRY_MASS_ERASE``. If this option is enabled, then flashing
+   only MCUBoot, for example using ``west flash --domain mcuboot``, may erase
+   the entire flash, including the main application image.
+
+Sysbuild Kconfig file
+*********************
+
+You can set sysbuild's Kconfig options for a single application using
+configuration files. By default, sysbuild looks for a configuration file named
+``sysbuild.conf`` in the application top-level directory, like this:
+
+.. code-block:: none
+
+   <home>/application
+   ├── CMakeLists.txt
+   ├── prj.conf
+   └── sysbuild.conf
+
+Here is an :file:`sysbuild.conf` file which enables building and flashing with
+MCUboot whenever sysbuild is used:
+
+.. code-block:: none
+
+   SB_CONFIG_BOOTLOADER_MCUBOOT=y
```

Comment:
```suggestion
.. code-block:: none

   SB_CONFIG_BOOTLOADER_MCUBOOT=y
```

Is there an introductory sentence for describing this Kconfig option?

Reply:
Good point, there doesn't seem to be such a sentence.

Let me try to find a place for introducing some sysbuild config option.